### PR TITLE
Add google tag manager to base layout

### PR DESCRIPTION
**Pull request by hexagoncircle**
This PR sets up the scripts necessary to connect google tag manager to this site.

### CODE REVIEW

**Review by stephan1echung — COMMENTED** [review]
Thanks @hexagoncircle ! Is it possible to have the tag come immediately after the `<head>` element?

I'm not seeing this GA4 tag on the https://deploy-preview-900--jamstack-site.netlify.app/conf/* pages.

**Comment by hexagoncircle** [comment]
> I'm not seeing this GA4 tag on the https://deploy-preview-900--jamstack-site.netlify.app/conf/* pages.

That's a different codebase. We'd need to add the scripts over there in a separate PR

File: src/site/_includes/layouts/base.njk [hunk 1]
```diff
@@ -9,6 +9,14 @@ ogimage: "/img/og/default-og-image.png"
...
     <meta charset="utf-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <!-- Google tag (gtag.js) -->
+    <script async src="https://www.googletagmanager.com/gtag/js?id=G-X2FMMZSSS9"></script>
+    <script>
+      window.dataLayer = window.dataLayer || [];
+      function gtag(){dataLayer.push(arguments);}
+      gtag('js', new Date());
+      gtag('config', 'G-X2FMMZSSS9');
+    </script>
```
Comment (stephan1echung) [inline src/site/_includes/layouts/base.njk:19]: @hexagoncircle In the [StackOverflow code example](https://stackoverflow.com/questions/46064817/how-high-in-the-head-section-can-you-safely-put-the-google-tag-manager-code/46065442#46065442), the google tag sits under the title and meta description. Did we want to move it under there as well?

Comment (hexagoncircle) [reply src/site/_includes/layouts/base.njk:25]: Sure, just pushed that update

**Review by stephan1echung — APPROVED** [review]
Thanks @hexagoncircle , LGTM!

**Pull request merged**